pkg/webservice/handlers_test.go

```diff
@@ -1690,6 +1694,28 @@ func TestGetApplicationHandler(t *testing.T) {
 	assert.Equal(t, http.StatusBadRequest, resp.statusCode, statusCodeError)
 	assert.Equal(t, errInfo.Message, "invalid URL escape \"%Zt\"", jsonMessageError)
 	assert.Equal(t, errInfo.StatusCode, http.StatusBadRequest)
+
+	// test additional application details
+	var req6 *http.Request
+	req6, err = http.NewRequest("GET", "/ws/v1/partition/default/queue/root.default/application/app-1", strings.NewReader(""))
+	assert.NilError(t, err, "HTTP request create failed")
+	req6 = req6.WithContext(context.WithValue(req.Context(), httprouter.ParamsKey, httprouter.Params{
+		httprouter.Param{Key: "partition", Value: partitionNameWithoutClusterID},
+		httprouter.Param{Key: "queue", Value: "root.default"},
+		httprouter.Param{Key: "application", Value: "app-1"},
+	}))
+	assert.NilError(t, err, "Get Application Handler request failed")
+	resp6 := &MockResponseWriter{}
+	var appDao *dao.ApplicationDAOInfo
+	getApplication(resp6, req6)
+	appSummary := app.GetApplicationSummary(partitionNameWithoutClusterID)
+	err = json.Unmarshal(resp6.outputBytes, &appDao)
+	assert.NilError(t, err, unmarshalError)
+	assert.Equal(t, "app-1", appDao.ApplicationID)
+	assert.Equal(t, app.StartTime().UnixMilli(), appDao.StartTime)
+	assert.Equal(t, appSummary.ResourceUsage.String(), appDao.ResourceUsage.String())
+	assert.Equal(t, appSummary.PreemptedResource.String(), appDao.PreemptedResource.String())
+	assert.Equal(t, appSummary.PlaceholderResource.String(), appDao.PlaceholderResource.String())
```

Comment:
Use `resources.Equals()` to verify equality

Reply:
@pbacsko (background: I am collaborating with @suprjinx on this PR.) I just added the `//nolint:funlen` as you requested. I attempted to use `resources.Equals()` on those 3 checks for `ResourceUsage`, `PreemptedResource`, and `PlaceholderResource`, however `resources.Equals()` takes two `*resource.Resource` arguments, and these 3 items are actually `*resources.TrackedResource`, which the compiler won't allow.  I wasn't able to implement an analogous `Equals(a, b *TrackedResource)` func, since the `resources` package already has `Equals(a, b *Resource)` and both `Resource` and `TrackedResource` are in the same `resources` package. So, should I make a `EqualsTrackedResource()` func, and rename the first to `EqualsResource()`, or just stay with the current test logic of `String()` comparison? Thanks.

Reply:
@richscott yes it's a different type I just found out...

I'd be happier with sth like `EqualsTracked()` created for `TrackedResource`. You can put it inside `tracked_resource.go`. The first layer is the extra code, the rest can be taken from the existing `Equals()`.

Eg.
```
func EqualsTracked(left, right *TrackedResource) bool {
	if left == right {
		return true
	}

	if left == nil || right == nil {
		return false
	}

	equals := true
	for k, v := range left.TrackedResourceMap {
		inner, ok := right.TrackedResourceMap[k]
		if !ok {
			return false
		}

		equals = equals && equalsMapContents(v, inner)
	}

	return equals
}

// basically the same as Equals()
func equalsMapContents(left, right map[string]int64) bool {
	if left == nil || right == nil {  // probably this is not needed
		return false
	}

	for k, v := range left {
		if right[k] != v {
			return false
		}
	}

	for k, v := range right {
		if left[k] != v {
			return false
		}
	}

	return true
}
```

I haven't tried this, but the main point is the approach. It would be much better if `TrackedResourceMap` was actually `map[string]*Resource` because we could re-use `Equals()` but this can be improved in a follow-up JIRA.

Reply:
@pbacsko  - updated as suggested - thanks for the code sample. I agree that adjusting `TrackedResourceMap` to use `map[string]*Resource` would be a nice clean-up in the near future. Thanks.

Reply:
I see that Codecov is failing its check - I will add a test for `EqualsTracked(..)` (mostly based off the `TestEquals(t)` in `resources_test.go`) to get the coverage up a bit so it passes.